### DISCUSSION TITLE
GitHub recovery codes usage

### DISCUSSION THREAD

**Discussion by ed-kurlyak** [community/community#116786] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have 2FA enabled on my account. In my everyday life I use a push-button telephone. But I also have a smartphone - it is lay turned off. To log into GitHub, I use recovery codes from the file github-recovery-codes.txt, I noticed that the codes that I already used once are no longer suitable for logging in. That is, I enter the code for ligin, and the next time this code for login no longer works. What is the problem? I may exhaust all the codes in the github-recovery-codes.txt file and lose access to GitHub.

**Comment by giovannivicentin** [comment]
Hello ed-kurlyak,

It's great to hear that you're using two-factor authentication (2FA) on your GitHub account as it adds an extra layer of security. To address your concern, the behavior you're experiencing with the recovery codes is actually by design. Recovery codes are meant to be used once and only once. Each code can help you sign in one time if you don't have access to your mobile device for 2FA. After a code is used, it becomes invalid, and you should strike it out from your list to avoid confusion.

Here are some steps to manage your recovery codes and ensure continued access to your GitHub account:

1. **Keep Track of Used Codes**: As you use a recovery code, make sure to mark it as used. It's essential to keep track so you don't attempt to use the same code twice.

2. **Generate New Codes**: Once you've used all of your recovery codes, or if you're running low, you should generate a new set of recovery codes as soon as possible. You can do this by logging into GitHub, going to the settings, and finding the section for two-factor authentication.

3. **Store Codes Securely**: Save the new set of recovery codes in a secure location. You might want to print them out and store them somewhere safe, or use a secure digital notes service.

4. **Access With Secondary Devices**: Since you have a smartphone that is usually turned off, consider using it as a secondary method for 2FA. You can use TOTP (Time-based One-Time Password) apps like Google Authenticator or Authy. These apps do not require a network connection to generate codes.

5. **Multiple Devices for 2FA**: If possible, set up another device that you keep on as a backup for 2FA. Some 2FA methods allow multiple devices to be prepared for generating codes.

6. **Avoid Lockout**: Always ensure that you have a backup authentication method set up in case you lose access to your primary 2FA device. GitHub allows you to set up multiple methods, like SMS, authentication apps, and security keys.

Remember, the recovery codes are your backup keys to your GitHub account. Protect them as you would your password, and ensure you generate a new set before you run out. This way, you can avoid being locked out of your account. If you ever feel that you're at risk of losing access, take a moment to set up your 2FA again with your smartphone or another device that you can keep handy.

Best regards

**Comment by ed-kurlyak** [comment]
Thank you, I generated new set recovery codes. But have problem - I use Microsoft Authenticator and have error using this app. See screenshot below.
![black4](https://github.com/community/community/assets/158138790/dd7c7695-b223-4330-9550-d48ea457db52)

**Reply by User132New** [reply]
You can use google authenticator, available on Play store!
Or use can use mobile OTP.
Or keep generating token before they expire.

**Comment by ed-kurlyak** [comment]
Always I used Microsoft Authenticator and work fine, but now not.

**Reply by User132New** [reply]
try reinstall or set up the authenticator again!
